jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -0,0 +1,272 @@
+  @Nullable private ImmutableList<String> programArguments;
+
+  /** Instantiate with {@link Jib#from}. */
+  JibContainerBuilder(ImageReference baseImageReference) {
+    this.baseImageReference = baseImageReference;
+  }
+
+  /**
+   * Adds a new layer to the container with {@code files} as the source files and {@code
+   * pathInContainer} as the path to copy the source files to in the container file system.
+   *
+   * <p>Source files that are directories will be recursively copied. For example, if the source
+   * files are:
+   *
+   * <ul>
+   *   <li>{@code fileA}
+   *   <li>{@code fileB}
+   *   <li>{@code directory/}
+   * </ul>
+   *
+   * and the destination to copy to is {@code /path/in/container}, then the new layer will have the
+   * following entries for the container file system:
+   *
+   * <ul>
+   *   <li>{@code /path/in/container/fileA}
+   *   <li>{@code /path/in/container/fileB}
+   *   <li>{@code /path/in/container/directory/}
+   *   <li>{@code /path/in/container/directory/...} (all contents of {@code directory/})
+   * </ul>
+   *
+   * @param files the source files to copy to a new layer in the container
+   * @param pathInContainer the Unix-style path to copy the source files to in the container file
+   *     system
+   * @return this
+   */
+  public JibContainerBuilder addLayer(List<Path> files, String pathInContainer) {
+    addLayer(LayerConfiguration.builder().addEntry(files, pathInContainer).build());
+    return this;
+  }
+
+  /**
+   * Sets the layers (defined by a list of {@link LayerConfiguration}s). This replaces any
+   * previously-added layers.
+   *
+   * @param layerConfigurations the list of {@link LayerConfiguration}s
+   * @return this
+   */
+  public JibContainerBuilder setLayers(List<LayerConfiguration> layerConfigurations) {
+    this.layerConfigurations = new ArrayList<>(layerConfigurations);
+    return this;
+  }
+
+  /**
+   * Sets the layers. This replaces any previously-added layers.
+   *
+   * @param layerConfigurations the {@link LayerConfiguration}s
+   * @return this
+   */
+  public JibContainerBuilder setLayers(LayerConfiguration... layerConfigurations) {
+    setLayers(Arrays.asList(layerConfigurations));
+    return this;
+  }
+
+  /**
+   * Adds a layer (defined by a {@link LayerConfiguration}).
+   *
+   * @param layerConfiguration the {@link LayerConfiguration}
+   * @return this
+   */
+  public JibContainerBuilder addLayer(LayerConfiguration layerConfiguration) {
+    layerConfigurations.add(layerConfiguration);
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint. This is the beginning of the command that is run when the
+   * container starts. {@link #setProgramArguments} sets additional tokens.
+   *
+   * @param entrypointTokens a list of tokens for the entrypoint command
+   * @return this
+   */
+  public JibContainerBuilder setEntrypoint(List<String> entrypointTokens) {
+    entrypoint = ImmutableList.copyOf(entrypointTokens);
```

Comment:
_token_ seems funny.  Did you choose it for a reason?  _component_?

Reply:
I feel like `entrypoint` might be fine here?

Reply:
Okay will remove tokens